doc/helpers/responsive-extension.md

```diff
@@ -0,0 +1,127 @@
+---
+uid: Toolkit.UI.ResponsiveExtension
+---
+# ResponsiveExtension
+
+## Summary
+The `ResponsiveExtension` class is a markup extension in the Uno Toolkit UI library that allows you to adapt the properties of a UI element based on the current window width.
+This is particularly useful for creating responsive designs that look good on a variety of screen sizes.
+
+### C#
+```csharp
+public partial class ResponsiveExtension : MarkupExtension, IResponsiveCallback
+```
+
+### XAML
+```xml
+xmlns:utu="using:Uno.Toolkit.UI"
+...
+
+<utu:ResponsiveExtension />
+```
+
+### Inheritance 
+Object &#8594; MarkupExtension &#8594; ResponsiveExtension
+
+### Constructors
+| Constructor           | Description                                                    |
+|-----------------------|----------------------------------------------------------------|
+| ResponsiveExtension() | Initializes a new instance of the `ResponsiveExtension` class. |
+
+### Properties
+| Property   | Type             | Description                                                                                         |
+| ---------- | ---------------- | --------------------------------------------------------------------------------------------------- |
+| Narrowest  | object           | Value to be used when the screen size is at its narrowest.                                          |
+| Narrow     | object           | Value to be used when the screen size is narrow.                                                    |
+| Normal     | object           | Value to be used when the screen size is normal.                                                    |
+| Wide       | object           | Value to be used when the screen size is wide.                                                      |
+| Widest     | object           | Value to be used when the screen size is at its widest.                                             |
+| Layout     | ResponsiveLayout | Overrides the MaxWidth for each screen size: `Narrowest`, `Narrow`, `Normal`, `Wide`, and `Widest`. |
+
+### Usage
+```xml
+<!-- Include the following XAML namespace to use the sample below -->
+xmlns:utu="using:Uno.Toolkit.UI"
+...
+<Page.Resources>
+	<SolidColorBrush x:Key="NarrowRed">Crimson</SolidColorBrush>
+	<SolidColorBrush x:Key="WideBlue">Blue</SolidColorBrush>
+</Page.Resources>
+...
+<Border Width="30"
+		Height="30"
+		HorizontalAlignment="Left"
+		Background="{utu:Responsive Normal={StaticResource NarrowRed},
+									Wide={StaticResource WideBlue}}" />
+```
```

Comment:
```suggestion
```xml
xmlns:utu="using:Uno.Toolkit.UI"
...

<Page.Resources>
	<SolidColorBrush x:Key="NarrowRed">Crimson</SolidColorBrush>
	<SolidColorBrush x:Key="WideBlue">Blue</SolidColorBrush>
</Page.Resources>
...

<TextBlock Text="Asd"
		   Background="{utu:Responsive Normal={StaticResource NarrowRed}, Wide={StaticResource WideBlue}}" />
```
```

Reply:
you should also add a remark here, that for anything other that `string`, you will need to properly "type" the value.
eg: Background={Responsive Normal=SkyBlue, Wide=Pink} will be an incorrect usage

Reply:
updated the section